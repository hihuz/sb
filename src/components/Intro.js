import React from "react";
import LinkBar from "./LinkBar";

const Intro = ({ title, text, greetings, links }) => (
  <section className="sb-intro">
    <article className="intro">
      <h1 className="intro-title">{title}</h1>
      <div className="logo-container">
        <img src="hihuz.jpg" alt="hihuz" />
      </div>
      {text.map(para => {
        const markup = { __html: para };
        return <h4 className="intro-text" key={para} dangerouslySetInnerHTML={markup} />;
      })}
      <h4 className="intro-text">{greetings}</h4>
      <LinkBar links={links} />
    </article>
  </section>
);

export default Intro;
