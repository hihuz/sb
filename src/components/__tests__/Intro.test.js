import React from "react";
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Intro from "../Intro";

describe("Intro", () => {
  test("Snapshot", () => {
    const component = shallow(<Intro text={[]} />);
    const tree = shallowToJson(component);
    expect(tree).toMatchSnapshot();
  });

  test("Should display the text passed as title props", () => {
    const title = "Hello";
    const component = shallow(<Intro title={title} text={[]} />);
    const expected = title;
    const actual = component.find(".intro-title").text();
    expect(actual).toEqual(expected);
  });

  test("Should render an h4 tags for each item in the text prop array and one for greetings", () => {
    const text = ["Hello", "World", "!"];
    const component = shallow(<Intro text={text} />);
    const expected = text.length + 1;
    const actual = component.find("h4").length;
    expect(actual).toEqual(expected);
  });

  test("Should display the text passed as greetings props in the last h4", () => {
    const text = ["Hello", "World!"];
    const greetings = "Wooo";
    const component = shallow(<Intro text={text} greetings={greetings} />);
    const expected = greetings;
    const actual = component.find("h4").last().text();
    expect(actual).toEqual(expected);
  });
});
