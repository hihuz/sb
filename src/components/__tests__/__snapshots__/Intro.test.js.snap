// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Intro Snapshot 1`] = `
<section
  className="sb-intro"
>
  <article
    className="intro"
  >
    <h1
      className="intro-title"
    />
    <div
      className="logo-container"
    >
      <img
        alt="hihuz"
        src="hihuz.jpg"
      />
    </div>
    <h4
      className="intro-text"
    />
    <LinkBar />
  </article>
</section>
`;
